x-pack/plugins/reporting/server/routes/lib/jobs_query.ts

```diff
@@ -87,6 +87,7 @@ export function jobsQueryFactory(reportingCore: ReportingCore) {
         elasticsearchClient.search({ body, index: getIndex() })
       );
 
+      // FIXME: return the info in ReportApiJSON format;
```

Comment:
This file is responsible for handling API requests that query jobs. One thing that is clear from looking at the API functional tests, the data formats of the Reporting endpoints are inconsistent.

 - "generate" API: returns `ReportApiJSON` 
 - "list" API: returns `ReportDocument[]`
 - "info" API: returns `ReportSource` 
 
To make it consistent, all the APIs should return `ReportApiJSON` format.

Reply:
All of the FIXMEs are being handled in the next PR: https://github.com/elastic/kibana/pull/102833